packages/config/src/Config.ts

```diff
@@ -1,12 +1,118 @@
 import JsonFile, { JSONObject } from '@expo/json-file';
-import fs from 'fs-extra';
 import path from 'path';
 import slug from 'slugify';
 
-import { AppJSONConfig, ExpRc, ExpoConfig, PackageJSONConfig, ProjectConfig } from './Config.types';
+import {
+  AppJSONConfig,
+  ConfigContext,
+  ExpRc,
+  ExpoConfig,
+  PackageJSONConfig,
+  Platform,
+  ProjectConfig,
+} from './Config.types';
 import { ConfigError } from './Errors';
+import { findAndEvalConfig } from './getConfig';
+import { getRootPackageJsonPath, projectHasModule } from './Modules';
 import { getExpoSDKVersion } from './Project';
 
+/**
+ * Get all platforms that a project is currently capable of running.
+ *
+ * @param projectRoot
+ * @param exp
+ */
+function getSupportedPlatforms(
+  projectRoot: string,
+  exp: Pick<ExpoConfig, 'nodeModulesPath'>
+): Platform[] {
+  const platforms: Platform[] = [];
+  if (projectHasModule('react-native', projectRoot, exp)) {
+    platforms.push('ios', 'android');
+  }
+  if (projectHasModule('react-native-web', projectRoot, exp)) {
+    platforms.push('web');
+  }
+  return platforms;
+}
+
+type GetConfigOptions = {
+  mode: 'development' | 'production';
+  configPath?: string;
+  skipSDKVersionRequirement?: boolean;
+};
+
+function getConfigContext(
+  projectRoot: string,
+  options: GetConfigOptions
+): { context: ConfigContext; pkg: JSONObject } {
+  // TODO(Bacon): This doesn't support changing the location of the package.json
+  const packageJsonPath = getRootPackageJsonPath(projectRoot, {});
+  const pkg = JsonFile.read(packageJsonPath);
+
+  const configPath = options.configPath || customConfigPaths[projectRoot];
+
+  // If the app.json exists, we'll read it and pass it to the app.config.js for further modification
+  const { configPath: appJsonConfigPath } = findConfigFile(projectRoot);
+  let rawConfig: JSONObject = {};
+  try {
+    rawConfig = JsonFile.read(appJsonConfigPath, { json5: true });
+  } catch (_) {}
+
+  const { exp: configFromPkg } = ensureConfigHasDefaultValues(projectRoot, rawConfig, pkg, true);
+
+  return {
+    pkg,
+    context: {
+      mode: options.mode,
+      projectRoot,
+      configPath,
+      config: configFromPkg,
+    },
+  };
+}
+
+/**
+ * Evaluate the config for an Expo project.
+ * If a function is exported from the `app.config.js` then a partial config will be passed as an argument.
+ * The partial config is composed from any existing app.json, and certain fields from the `package.json` like name and description.
+ *
+ * You should use the supplied `mode` option in an `app.config.js` instead of `process.env.NODE_ENV`
+ *
+ * **Example**
+ * ```js
+ * module.exports = function({ config, mode }) {
+ *   // mutate the config before returning it.
+ *   config.slug = 'new slug'
```

Comment:
It seems that the `config` object currently passed in doesn't have `config.slug`, but `config.expo.slug`?